specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json

```diff
@@ -5330,6 +5491,79 @@
           "description": "Error when What-If operation fails."
         }
       }
+    },
+    "Tags": {
+      "description": "key and value pairs for tags",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string",
+            "description": "The tag value."
+          }
+        }
+      }
+    },
+    "TagPatchRequest": {
+      "type": "object",
+      "description": "Tag Request for Patch operation.",
+      "properties": {
+        "operation": {
+          "type": "string",
+          "description": "The operation type for the patch api.",
+          "enum": [
+            "Replace",
+            "Merge",
+            "Delete"
+          ]
+        },
+        "properties": {
+          "$ref": "#/definitions/Tags",
+          "description": "tags object passing in the request."
+        }
+      }
+    },
+    "TagRequest": {
```

Comment:
I'm wondering if we can just collapse TagRequest and TagResponse into a single definition and use something like the following:

```
"TagsResource": {
  "type": "object",
  "description": "Tags resource.",
  "required": [
    "properties"
  ],
  "properties": {
    "properties": {
      "$ref": "#/definitions/Tags",
      "description": "tags object passing in the request."
    }
  },
  "allOf": [
    {
      "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/Resource"
    }
  ]
},
```